### PR TITLE
Add details on choosing first stage models

### PR DESCRIPTION
**Pull request by moprescu**
Modified docs, README and added a new notebook

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -416,6 +417,39 @@ See the <a href="#references">References</a> section for more details.
...
+                  "n_estimators": (10, 30, 50, 100, 200),
+                  "max_features": (2, 4, 6),
+              },
+              cv=5,
+           )
+# First stage model selection within EconML
+# This is slower, but more direct
+est = LinearDML(model_y=cv_model, model_t=cv_model)
+# First stage model selection ouside of EconML
+# Faster, but needs boilerplate code
```
Comment (vsyrgkanis) [inline README.md:446]: I feel we might want to take a stance and recommend this e.g.: 
as long as you are not selecting over a huge set of hyperparameter values, then hyperparameter tuning outside of econml is recommended both for statistical and computational reasons.

File: doc/spec/estimation/dml.rst [hunk 2]
```diff
@@ -430,19 +430,40 @@ Usage FAQs
...
                                 'max_depth': [3, None],
-                                'n_estimators': (10, 30, 50, 100, 200, 400, 600, 800, 1000),
+                                'n_estimators': (10, 30, 50, 100, 200),
                                 'max_features': (2,4,6)
                             }, cv=10, n_jobs=-1, scoring='neg_mean_squared_error'
                         )
         est = SparseLinearDML(model_y=first_stage(), model_t=first_stage())
 
+    Alternatively, you can pick the best first stage models outside of the EconML framework and pass in the selected models to EconML. 
+    This can save on runtime and computational resources. E.g.:
```
Comment (vsyrgkanis) [inline doc/spec/estimation/dml.rst:447]: Also add that ti can be staitstically more stalbe as all data are used for hyperparmeter tuning as opposed to a single folds data.

File: notebooks/Choosing First Stage Models.ipynb [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "model_y = first_stage().fit(XW, Y).best_estimator_\n",
+    "model_t = first_stage().fit(XW, T).best_estimator_\n",
+    "est = LinearDML(\n",
```
Comment (vsyrgkanis) [inline notebooks/Choosing First Stage Models.ipynb:561]: MAybe set `linear_first_stages=False` throughout the notebook.

File: notebooks/Choosing First Stage Models.ipynb [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.plot(X_test, test_effect, \"--\", label=\"Truth\")\n",
```
Comment (vsyrgkanis) [inline notebooks/Choosing First Stage Models.ipynb:611]: Weird that the final approach that does a CV among both lasso and forest is worse than just lasso. Doesn't seem right.

File: notebooks/Choosing First Stage Models.ipynb [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Performing first stage model selection outside of EconML\n",
+    "\n",
+    "An alternative to passing in CV models to EconML is to perform model selection outside of the EconML estimators and then pass in the pre-selected models to EconML.\n",
+    "\n",
+    "**Advantages:** Faster runtimes of the EconML estimators and more flexible selection of first stage models.\n",
```
Comment (vsyrgkanis) [inline notebooks/Choosing First Stage Models.ipynb:292]: Add here also that as long as log(number of hyper-parameters) is much smaller than the number of samples, this approach maintains statisical validity of the resulting inference results.

**Review by vsyrgkanis — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (moprescu) [reply README.md]: done!

File: doc/spec/estimation/dml.rst (hunk 2 above)
Comment (moprescu) [reply doc/spec/estimation/dml.rst]: Done!

File: notebooks/Choosing First Stage Models.ipynb (hunk 3 above)
Comment (moprescu) [reply notebooks/Choosing First Stage Models.ipynb:569]: Done!

**Pull request merged**